canonical-web-and-design/maas-ui
# feat(ui): edit machine name from the details header

## Done

- Add the form for editing the machine name from the machine details header.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View a locked machine and click on the machine name, nothing should happen.
- View a machine you can edit and click on the machine name, the form should appear.
- Edit the machine name or domain and click Save, it should save the details.

## Fixes

Fixes: canonical-web-and-design/maas-squad#2102.
